In [1]:
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np
import seaborn as sns

In [2]:
df = pd.read_csv('../dataset/OnlineNewsPopularity.csv')
df.columns = df.columns.str.strip()

In [3]:
df.head()

url  timedelta  \
0  http://mashable.com/2013/01/07/amazon-instant-...      731.0   
1  http://mashable.com/2013/01/07/ap-samsung-spon...      731.0   
2  http://mashable.com/2013/01/07/apple-40-billio...      731.0   
3  http://mashable.com/2013/01/07/astronaut-notre...      731.0   
4   http://mashable.com/2013/01/07/att-u-verse-apps/      731.0   

   n_tokens_title  n_tokens_content  n_unique_tokens  n_non_stop_words  \
0            12.0             219.0         0.663594               1.0   
1             9.0             255.0         0.604743               1.0   
2             9.0             211.0         0.575130               1.0   
3             9.0             531.0         0.503788               1.0   
4            13.0            1072.0         0.415646               1.0   

   n_non_stop_unique_tokens  num_hrefs  num_self_hrefs  num_imgs  ...  \
0                  0.815385        4.0             2.0       1.0  ...   
1                  0.791946        3.0             1.0       1.0  ...   
2                  0.663866        3.0             1.0       1.0  ...   
3                  0.665635        9.0             0.0       1.0  ...   
4                  0.540890       19.0            19.0      20.0  ...   

   min_positive_polarity  max_positive_polarity  avg_negative_polarity  \
0               0.100000                    0.7              -0.350000   
1               0.033333                    0.7              -0.118750   
2               0.100000                    1.0              -0.466667   
3               0.136364                    0.8              -0.369697   
4               0.033333                    1.0              -0.220192   

   min_negative_polarity  max_negative_polarity  title_subjectivity  \
0                 -0.600              -0.200000            0.500000   
1                 -0.125              -0.100000            0.000000   
2                 -0.800              -0.133333            0.000000   
3                 -0.600              -0.166667            0.000000   
4                 -0.500              -0.050000            0.454545   

   title_sentiment_polarity  abs_title_subjectivity  \
0                 -0.187500                0.000000   
1                  0.000000                0.500000   
2                  0.000000                0.500000   
3                  0.000000                0.500000   
4                  0.136364                0.045455   

   abs_title_sentiment_polarity  shares  
0                      0.187500     593  
1                      0.000000     711  
2                      0.000000    1500  
3                      0.000000    1200  
4                      0.136364     505  

[5 rows x 61 columns]

In [4]:
# Dataset basic information
print("Filas, columnas:", df.shape)
print("\nColumnas:")
print(df.columns.tolist())
print("\nTipos de datos:")
print(df.dtypes.value_counts())
print("\nValores faltantes:")
print(df.isnull().sum().sum())
print("\nEstadísticas básicas:")
df.describe()

Filas, columnas: (39644, 61)

Columnas:
['url', 'timedelta', 'n_tokens_title', 'n_tokens_content', 'n_unique_tokens', 'n_non_stop_words', 'n_non_stop_unique_tokens', 'num_hrefs', 'num_self_hrefs', 'num_imgs', 'num_videos', 'average_token_length', 'num_keywords', 'data_channel_is_lifestyle', 'data_channel_is_entertainment', 'data_channel_is_bus', 'data_channel_is_socmed', 'data_channel_is_tech', 'data_channel_is_world', 'kw_min_min', 'kw_max_min', 'kw_avg_min', 'kw_min_max', 'kw_max_max', 'kw_avg_max', 'kw_min_avg', 'kw_max_avg', 'kw_avg_avg', 'self_reference_min_shares', 'self_reference_max_shares', 'self_reference_avg_sharess', 'weekday_is_monday', 'weekday_is_tuesday', 'weekday_is_wednesday', 'weekday_is_thursday', 'weekday_is_friday', 'weekday_is_saturday', 'weekday_is_sunday', 'is_weekend', 'LDA_00', 'LDA_01', 'LDA_02', 'LDA_03', 'LDA_04', 'global_subjectivity', 'global_sentiment_polarity', 'global_rate_positive_words', 'global_rate_negative_words', 'rate_positive_words', 'rate_neg

timedelta  n_tokens_title  n_tokens_content  n_unique_tokens  \
count  39644.000000    39644.000000      39644.000000     39644.000000   
mean     354.530471       10.398749        546.514731         0.548216   
std      214.163767        2.114037        471.107508         3.520708   
min        8.000000        2.000000          0.000000         0.000000   
25%      164.000000        9.000000        246.000000         0.470870   
50%      339.000000       10.000000        409.000000         0.539226   
75%      542.000000       12.000000        716.000000         0.608696   
max      731.000000       23.000000       8474.000000       701.000000   

       n_non_stop_words  n_non_stop_unique_tokens     num_hrefs  \
count      39644.000000              39644.000000  39644.000000   
mean           0.996469                  0.689175     10.883690   
std            5.231231                  3.264816     11.332017   
min            0.000000                  0.000000      0.000000   
25%            1.000000                  0.625739      4.000000   
50%            1.000000                  0.690476      8.000000   
75%            1.000000                  0.754630     14.000000   
max         1042.000000                650.000000    304.000000   

       num_self_hrefs      num_imgs    num_videos  ...  min_positive_polarity  \
count    39644.000000  39644.000000  39644.000000  ...           39644.000000   
mean         3.293638      4.544143      1.249874  ...               0.095446   
std          3.855141      8.309434      4.107855  ...               0.071315   
min          0.000000      0.000000      0.000000  ...               0.000000   
25%          1.000000      1.000000      0.000000  ...               0.050000   
50%          3.000000      1.000000      0.000000  ...               0.100000   
75%          4.000000      4.000000      1.000000  ...               0.100000   
max        116.000000    128.000000     91.000000  ...               1.000000   

       max_positive_polarity  avg_negative_polarity  min_negative_polarity  \
count           39644.000000           39644.000000           39644.000000   
mean                0.756728              -0.259524              -0.521944   
std                 0.247786               0.127726               0.290290   
min                 0.000000              -1.000000              -1.000000   
25%                 0.600000              -0.328383              -0.700000   
50%                 0.800000              -0.253333              -0.500000   
75%                 1.000000              -0.186905              -0.300000   
max                 1.000000               0.000000               0.000000   

       max_negative_polarity  title_subjectivity  title_sentiment_polarity  \
count           39644.000000        39644.000000              39644.000000   
mean               -0.107500            0.282353                  0.071425   
std                 0.095373            0.324247                  0.265450   
min                -1.000000            0.000000                 -1.000000   
25%                -0.125000            0.000000                  0.000000   
50%                -0.100000            0.150000                  0.000000   
75%                -0.050000            0.500000                  0.150000   
max                 0.000000            1.000000                  1.000000   

       abs_title_subjectivity  abs_title_sentiment_polarity         shares  
count            39644.000000                  39644.000000   39644.000000  
mean                 0.341843                      0.156064    3395.380184  
std                  0.188791                      0.226294   11626.950749  
min                  0.000000                      0.000000       1.000000  
25%                  0.166667                      0.000000     946.000000  
50%                  0.500000                      0.000000    1400.000000  
75%                  0.500000                      0.250000    2

El dataset no tiene valores nulos, no se necesita imputación.

In [5]:
pd.set_option('display.max_colwidth', None)
pd.set_option('display.width',  None)

print("=== Valores extremos ===")
print("n_unique_tokens > 1:")
print(df[df['n_unique_tokens'] > 1][['url',
'n_unique_tokens', 'n_tokens_content']].head())

print("\n=== Inconsistencias lógicas ===")
print("Artículos con 0 tokens de contenido pero otras métricas:")
zero_content = df[df['n_tokens_content'] == 0]
print(f"Count: {len(zero_content)}")
if len(zero_content) > 0:
    print(zero_content[['url', 'n_tokens_content',
'num_hrefs', 'num_imgs']].head())

print("\n=== DISTRIBUCIONES EXTREMAS ===")
print("Shares > 100k:")
print(f"Count: {len(df[df['shares'] > 100000])}")

print("\nnum_hrefs > 100:")
print(f"Count: {len(df[df['num_hrefs'] > 100])}")

print(f"\nn_unique_tokens range:  {df['n_unique_tokens'].min():.3f} to {df['n_unique_tokens'].max():.3f}")
print(f"Expected range for ratios: 0 to 1")


=== Valores extremos ===
n_unique_tokens > 1:
                                                                    url  \
31037  http://mashable.com/2014/08/18/ukraine-civilian-convoy-attacked/   

       n_unique_tokens  n_tokens_content  
31037            701.0            1570.0  

=== Inconsistencias lógicas ===
Artículos con 0 tokens de contenido pero otras métricas:
Count: 1181
                                                                     url  \
893       http://mashable.com/2013/01/23/actual-facebook-graph-searches/   
917   http://mashable.com/2013/01/23/fitness-gadget-gym-cost-comparison/   
1062          http://mashable.com/2013/01/25/data-vs-nature-infographic/   
1121              http://mashable.com/2013/01/26/infographics-marketing/   
1312                http://mashable.com/2013/01/29/social-tv-chart-1-29/   

      n_tokens_content  num_hrefs  num_imgs  
893                0.0        0.0      10.0  
917                0.0        0.0       0.0  
1062               0

Revisamos esos artículos uno por uno y constatamos que son artículos mal registrados. El que tiene 701 tokens únicos es un artículo normal, es claramente un error de registro. Lo mismo pasa con las páginas con 0 tokens de contenido: todas tienen contenido habitual. 

In [6]:
# Identificar todas las columnas que deberían ser ratios (0-1)
ratio_columns = [
    'n_unique_tokens',           # ratio de tokens únicos
    'n_non_stop_words',          # ratio de palabras no-stop
    'n_non_stop_unique_tokens',  # ratio de tokens únicos no-stop
    'rate_positive_words',       # ratio de palabras positivas
    'rate_negative_words',       # ratio de palabras negativas
    'global_subjectivity',       # subjetividad global
    'global_sentiment_polarity', # polaridad de sentimiento global
    'global_rate_positive_words', # ratio global de palabras positivas
    'global_rate_negative_words', # ratio global de palabras negativas
    'avg_positive_polarity',     # polaridad positiva promedio
    'min_positive_polarity',     # polaridad positiva mínima
    'max_positive_polarity',     # polaridad positiva máxima
    'avg_negative_polarity',     # polaridad negativa promedio
    'min_negative_polarity',     # polaridad negativa mínima
    'max_negative_polarity',     # polaridad negativa máxima
    'title_subjectivity',        # subjetividad del título
    'title_sentiment_polarity',  # polaridad del sentimiento del título
    'abs_title_subjectivity',    # subjetividad absoluta del título
    'abs_title_sentiment_polarity' # polaridad absoluta del sentimiento del título
]

print("=== COLUMNAS CON CORRUPCIÓN DE RATIOS ===")
corrupted_ratios = []

for col in ratio_columns:
    min_val = df[col].min()
    max_val = df[col].max()
    out_of_range = len(df[(df[col] < 0) | (df[col] > 1)])
    
    if out_of_range >= 1:
        corrupted_ratios.append(col)
        print(f"{col}:")
        print(f"  Rango: {min_val:.3f} a {max_val:.3f}")
        print(f"  Fuera de [0,1]: {out_of_range} casos")
        if out_of_range <= 10:
            mask = (df[col] < 0) | (df[col] > 1)
            extreme_values = df.loc[mask, col].values
            print(f"  Valores extremos: {sorted(extreme_values)}")
        else:
            print(f"  Top 5 valores: {df[col].nlargest(5).values}")
        print()

print(f"Total columnas corruptas: {len(corrupted_ratios)}")
print(f"Columnas: {corrupted_ratios}")

=== COLUMNAS CON CORRUPCIÓN DE RATIOS ===
n_unique_tokens:
  Rango: 0.000 a 701.000
  Fuera de [0,1]: 1 casos
  Valores extremos: [np.float64(701.0)]

n_non_stop_words:
  Rango: 0.000 a 1042.000
  Fuera de [0,1]: 1 casos
  Valores extremos: [np.float64(1042.0)]

n_non_stop_unique_tokens:
  Rango: 0.000 a 650.000
  Fuera de [0,1]: 1 casos
  Valores extremos: [np.float64(650.0)]

global_sentiment_polarity:
  Rango: -0.394 a 0.728
  Fuera de [0,1]: 3264 casos
  Top 5 valores: [0.72784091 0.655      0.65       0.63174603 0.625     ]

avg_negative_polarity:
  Rango: -1.000 a 0.000
  Fuera de [0,1]: 37094 casos
  Top 5 valores: [0. 0. 0. 0. 0.]

min_negative_polarity:
  Rango: -1.000 a 0.000
  Fuera de [0,1]: 37094 casos
  Top 5 valores: [0. 0. 0. 0. 0.]

max_negative_polarity:
  Rango: -1.000 a 0.000
  Fuera de [0,1]: 37094 casos
  Top 5 valores: [0. 0. 0. 0. 0.]

title_sentiment_polarity:
  Rango: -1.000 a 1.000
  Fuera de [0,1]: 5786 casos
  Top 5 valores: [1. 1. 1. 1. 1.]

Total columnas

Identificamos que una fila tiene valores absolutos en n_unique_tokens, n_non_stop_words, n_non_stop_unique_tokens, columnas que usan ratios (decimales de 0 a 1). Por lo tanto, decidimos transformarlas al ratio correspondiente.

In [7]:
# Corregir corrupción en ratios de tokens
print("=== CORRECCIÓN DE RATIOS CORRUPTOS ===")

# Identificar la fila corrupta
corrupt_idx = 31037
row = df.iloc[corrupt_idx]

print("Valores antes de la corrección:")
print(f"n_tokens_content: {row['n_tokens_content']}")
print(f"n_unique_tokens: {row['n_unique_tokens']}")
print(f"n_non_stop_words: {row['n_non_stop_words']}")
print(f"n_non_stop_unique_tokens: {row['n_non_stop_unique_tokens']}")

# Calcular ratios correctos
if row['n_tokens_content'] > 0:
    correct_unique_ratio = row['n_unique_tokens'] / row['n_tokens_content']
    correct_non_stop_ratio = row['n_non_stop_words'] / row['n_tokens_content'] 
    correct_non_stop_unique_ratio = row['n_non_stop_unique_tokens'] / row['n_tokens_content']
    
    print(f"\nRatios corregidos:")
    print(f"n_unique_tokens: {correct_unique_ratio:.6f}")
    print(f"n_non_stop_words: {correct_non_stop_ratio:.6f}")
    print(f"n_non_stop_unique_tokens: {correct_non_stop_unique_ratio:.6f}")
    
    # Aplicar correcciones
    df.loc[corrupt_idx, 'n_unique_tokens'] = correct_unique_ratio
    df.loc[corrupt_idx, 'n_non_stop_words'] = correct_non_stop_ratio
    df.loc[corrupt_idx, 'n_non_stop_unique_tokens'] = correct_non_stop_unique_ratio
    
    print(f"\n✓ Ratios corregidos exitosamente")
    
    # Verificar corrección
    print(f"\nVerificación post-corrección:")
    print(f"n_unique_tokens rango: {df['n_unique_tokens'].min():.6f} a {df['n_unique_tokens'].max():.6f}")
    print(f"n_non_stop_words rango: {df['n_non_stop_words'].min():.6f} a {df['n_non_stop_words'].max():.6f}")  
    print(f"n_non_stop_unique_tokens rango: {df['n_non_stop_unique_tokens'].min():.6f} a {df['n_non_stop_unique_tokens'].max():.6f}")
    
else:
    print("Error: no se puede calcular ratio con n_tokens_content = 0")

=== CORRECCIÓN DE RATIOS CORRUPTOS ===
Valores antes de la corrección:
n_tokens_content: 1570.0
n_unique_tokens: 701.0
n_non_stop_words: 1042.0
n_non_stop_unique_tokens: 650.0

Ratios corregidos:
n_unique_tokens: 0.446497
n_non_stop_words: 0.663694
n_non_stop_unique_tokens: 0.414013

✓ Ratios corregidos exitosamente

Verificación post-corrección:
n_unique_tokens rango: 0.000000 a 1.000000
n_non_stop_words rango: 0.000000 a 1.000000
n_non_stop_unique_tokens rango: 0.000000 a 1.000000


In [8]:
# Analizar artículos con 0 tokens de contenido
print("=== ANÁLISIS ARTÍCULOS CON 0 TOKENS ===")

zero_content = df[df['n_tokens_content'] == 0]
print(f"Total artículos con 0 tokens: {len(zero_content)}")
print(f"Porcentaje del dataset: {len(zero_content)/len(df)*100:.2f}%")

print(f"\nCaracterísticas de estos artículos:")
print(f"num_hrefs promedio: {zero_content['num_hrefs'].mean():.2f}")
print(f"num_imgs promedio: {zero_content['num_imgs'].mean():.2f}")
print(f"shares promedio: {zero_content['shares'].mean():.0f}")

print(f"\nDistribución de shares:")
print(zero_content['shares'].describe())

print(f"\nComparación con artículos normales:")
normal_content = df[df['n_tokens_content'] > 0]
print(f"Shares promedio (contenido > 0): {normal_content['shares'].mean():.0f}")
print(f"Shares promedio (contenido = 0): {zero_content['shares'].mean():.0f}")

print(f"\n¿Son artículos válidos para predicción?")
print(f"Artículos 0-contenido con shares > 1000: {len(zero_content[zero_content['shares'] > 1000])}")
print(f"Artículos 0-contenido con imágenes: {len(zero_content[zero_content['num_imgs'] > 0])}")

=== ANÁLISIS ARTÍCULOS CON 0 TOKENS ===
Total artículos con 0 tokens: 1181
Porcentaje del dataset: 2.98%

Características de estos artículos:
num_hrefs promedio: 0.00
num_imgs promedio: 3.93
shares promedio: 4699

Distribución de shares:
count      1181.000000
mean       4698.751058
std       12827.939697
min           4.000000
25%        1000.000000
50%        1600.000000
75%        3800.000000
max      211600.000000
Name: shares, dtype: float64

Comparación con artículos normales:
Shares promedio (contenido > 0): 3355
Shares promedio (contenido = 0): 4699

¿Son artículos válidos para predicción?
Artículos 0-contenido con shares > 1000: 853
Artículos 0-contenido con imágenes: 364


Tenemos un 2.98% de artículos con 0 tokens de contenido. Al ingresar a enlaces aleatoriamente detectamos que se tratan de artículos *con contenido* pero que fueron incorrectamente ingresados con valor 0. Decidimos eliminarlas porque representan un porcentaje poco significativo del total de datos, y no aportan valor por otras variables.

In [9]:
# Eliminar artículos con 0 tokens de contenido
print("=== ELIMINACIÓN DE ARTÍCULOS SIN CONTENIDO ===")

print(f"Dataset original: {len(df)} filas")
zero_content_count = len(df[df['n_tokens_content'] == 0])
print(f"Artículos a eliminar: {zero_content_count}")

# Eliminar filas con contenido = 0
df_clean = df[df['n_tokens_content'] > 0].copy()

print(f"Dataset limpio: {len(df_clean)} filas")
print(f"Filas eliminadas: {len(df) - len(df_clean)}")
print(f"Porcentaje retenido: {len(df_clean)/len(df)*100:.1f}%")

# Verificar eliminación
print(f"\nVerificación:")
print(f"Mínimo n_tokens_content: {df_clean['n_tokens_content'].min()}")
print(f"Artículos con 0 tokens restantes: {len(df_clean[df_clean['n_tokens_content'] == 0])}")

# Actualizar variable principal
df = df_clean
df.head()

=== ELIMINACIÓN DE ARTÍCULOS SIN CONTENIDO ===
Dataset original: 39644 filas
Artículos a eliminar: 1181
Dataset limpio: 38463 filas
Filas eliminadas: 1181
Porcentaje retenido: 97.0%

Verificación:
Mínimo n_tokens_content: 18.0
Artículos con 0 tokens restantes: 0


url  timedelta  \
0    http://mashable.com/2013/01/07/amazon-instant-video-browser/      731.0   
1     http://mashable.com/2013/01/07/ap-samsung-sponsored-tweets/      731.0   
2  http://mashable.com/2013/01/07/apple-40-billion-app-downloads/      731.0   
3        http://mashable.com/2013/01/07/astronaut-notre-dame-bcs/      731.0   
4                http://mashable.com/2013/01/07/att-u-verse-apps/      731.0   

   n_tokens_title  n_tokens_content  n_unique_tokens  n_non_stop_words  \
0            12.0             219.0         0.663594               1.0   
1             9.0             255.0         0.604743               1.0   
2             9.0             211.0         0.575130               1.0   
3             9.0             531.0         0.503788               1.0   
4            13.0            1072.0         0.415646               1.0   

   n_non_stop_unique_tokens  num_hrefs  num_self_hrefs  num_imgs  ...  \
0                  0.815385        4.0             2.0       1.0  ...   
1                  0.791946        3.0             1.0       1.0  ...   
2                  0.663866        3.0             1.0       1.0  ...   
3                  0.665635        9.0             0.0       1.0  ...   
4                  0.540890       19.0            19.0      20.0  ...   

   min_positive_polarity  max_positive_polarity  avg_negative_polarity  \
0               0.100000                    0.7              -0.350000   
1               0.033333                    0.7              -0.118750   
2               0.100000                    1.0              -0.466667   
3               0.136364                    0.8              -0.369697   
4               0.033333                    1.0              -0.220192   

   min_negative_polarity  max_negative_polarity  title_subjectivity  \
0                 -0.600              -0.200000            0.500000   
1                 -0.125              -0.100000            0.000000   
2                 -0.800              -0.133333            0.000000   
3                 -0.600              -0.166667            0.000000   
4                 -0.500              -0.050000            0.454545   

   title_sentiment_polarity  abs_title_subjectivity  \
0                 -0.187500                0.000000   
1                  0.000000                0.500000   
2                  0.000000                0.500000   
3                  0.000000                0.500000   
4                  0.136364                0.045455   

   abs_title_sentiment_polarity  shares  
0                      0.187500     593  
1                      0.000000     711  
2                      0.000000    1500  
3                      0.000000    1200  
4                      0.136364     505  

[5 rows x 61 columns]

In [10]:
# Análisis de outliers extremos
print("=== ANÁLISIS DE OUTLIERS EXTREMOS ===")

# 1. Shares extremas (>100k)
extreme_shares = df[df['shares'] > 100000]
print(f"1. SHARES > 100K:")
print(f"   Cantidad: {len(extreme_shares)}")
print(f"   Rango: {extreme_shares['shares'].min():,} a {extreme_shares['shares'].max():,}")
print(f"   Mediana: {extreme_shares['shares'].median():,}")

# Ver algunas características de artículos virales
print(f"\n   Características de artículos virales:")
print(f"   n_tokens_content promedio: {extreme_shares['n_tokens_content'].mean():.0f}")
print(f"   num_hrefs promedio: {extreme_shares['num_hrefs'].mean():.1f}")
print(f"   num_imgs promedio: {extreme_shares['num_imgs'].mean():.1f}")

# 2. hrefs extremos (>100)
extreme_hrefs = df[df['num_hrefs'] > 100]
print(f"\n2. HREFS > 100:")
print(f"   Cantidad: {len(extreme_hrefs)}")
print(f"   Rango: {extreme_hrefs['num_hrefs'].min()} a {extreme_hrefs['num_hrefs'].max()}")
print(f"   Shares promedio: {extreme_hrefs['shares'].mean():,.0f}")

print(f"\n   Características de artículos con muchos links:")
print(f"   n_tokens_content promedio: {extreme_hrefs['n_tokens_content'].mean():.0f}")
print(f"   shares vs normal: {extreme_hrefs['shares'].mean():.0f} vs {df['shares'].mean():.0f}")

# 3. Distribuciones generales usando IQR
print(f"\n3. ANÁLISIS IQR PARA THRESHOLDS:")

for col in ['shares', 'num_hrefs']:
    q1 = df[col].quantile(0.25)
    q3 = df[col].quantile(0.75)
    iqr = q3 - q1
    lower_bound = q1 - 1.5 * iqr
    upper_bound = q3 + 1.5 * iqr
    
    outliers = len(df[(df[col] < lower_bound) | (df[col] > upper_bound)])
    
    print(f"\n   {col}:")
    print(f"   Q1: {q1:.0f}, Q3: {q3:.0f}, IQR: {iqr:.0f}")
    print(f"   Threshold superior: {upper_bound:.0f}")
    print(f"   Outliers (IQR): {outliers} casos")
    print(f"   Max valor: {df[col].max():.0f}")

=== ANÁLISIS DE OUTLIERS EXTREMOS ===
1. SHARES > 100K:
   Cantidad: 52
   Rango: 100,300 a 843,300
   Mediana: 139,550.0

   Características de artículos virales:
   n_tokens_content promedio: 608
   num_hrefs promedio: 16.0
   num_imgs promedio: 6.9

2. HREFS > 100:
   Cantidad: 51
   Rango: 101.0 a 304.0
   Shares promedio: 7,488

   Características de artículos con muchos links:
   n_tokens_content promedio: 1804
   shares vs normal: 7488 vs 3355

3. ANÁLISIS IQR PARA THRESHOLDS:

   shares:
   Q1: 945, Q3: 2700, IQR: 1755
   Threshold superior: 5332
   Outliers (IQR): 4523 casos
   Max valor: 843300

   num_hrefs:
   Q1: 5, Q3: 14, IQR: 9
   Threshold superior: 28
   Outliers (IQR): 2636 casos
   Max valor: 304


Se observan valores razonables, propios de contenido viral. Se decide no eliminar ni transformar ningún dato.